docs/resource_assembly_migration.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Tiltfile Resource Assembly Migration
+layout: docs
+---
+
+Tilt is changing how it combines k8s objects and images into Tilt resources.
```

Comment:
do we have a definition of a "resource" that we can point to? otherwise, is it worth a parenthetical definition?

Reply:
ooh maybe link to that section in the Concepts doc?